modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/consensus/HealthCheckConsensus.scala

```diff
@@ -116,6 +117,10 @@ abstract class HealthCheckConsensus[
 
             ConsensusRounds(updatedHistorical, updatedInProgress)
         }
+      }.flatTap { _ =>
+        waitingProposals.modify { proposals =>
+          (Set.empty[B], proposals.toList.traverse(handleProposal(_)))
```

Comment:
Every time it's not processed for inProgress or historical round it will come back to the pool right?

Reply:
Yes, that's the current implementation